cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -2055,6 +2035,116 @@ class AzureFileSystem::Impl {
   static constexpr auto kTimeNeededForFileOrDirectoryRename = std::chrono::seconds{3};
 
  public:
+  /// \pre location.container is not empty.
+  /// \pre location.path is not empty.
+  /// \pre location.path does not end with a trailing slash.
+  Status DeleteFileOnFileSystem(const DataLake::DataLakeFileSystemClient& adlfs_client,
+                                const AzureLocation& location, bool require_file_to_exist,
+                                Azure::Nullable<std::string> lease_id = {}) {
```

Comment:
When do we need `lease_id`?

Reply:
This was needed for an experiment I did with the API before, but I'm removing this together with the implementation of `MovePaths` for flat-namespace accounts.

Reply:
I removed it in the latest commits.